crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -498,10 +545,29 @@ module sui::sui_system {
         ctx: &TxContext,
     ) {
         let self = load_system_state_mut(self);
-        let validator = validator_set::get_validator_mut_with_ctx(&mut self.validators, ctx);
+        let validator = validator_set::get_validator_mut_with_ctx(
+            &mut self.validators,
+            ctx,
+            false /* include_candidate */
+        );
         validator::update_next_epoch_network_address(validator, network_address);
     }
 
+    /// Update candidate validator's network address.
+    public entry fun update_candidate_validator_network_address(
+        self: &mut SuiSystemState,
+        network_address: vector<u8>,
+        ctx: &TxContext,
+    ) {
+        let self = load_system_state_mut(self);
+        let candidate = validator_set::get_validator_mut_with_ctx(
+            &mut self.validators,
```

Comment:
For these entry functions which can only be called by candidates, it's better to have a dedicted `get_validator_candidate_mut_with_ctx` so we can get rid of the boolean here?

Reply:
It's not as much as getting a candidate only, but rather to include the candidate in the search, so I renamed it to `get_validator_mut_with_ctx_including_candidates` but I am open to other suggestions.